lib/utilities/npm.js

```diff
@@ -1,38 +1,34 @@
 'use strict';
 
+var exec = require('child_process').exec;
 var Promise = require('../ext/promise');
+var SilentError = require('silent-error');
 
-//
+var debug = require('heimdalljs-logger')('ember-cli:npm');
 
 /**
-  Runs the npm command `command` with the supplied args and load options.
-
-  Please note that the loaded module appears to retain some state, so do not
-  expect multiple invocations within the same process to work without quirks.
-  This problem is likely fixable.
+  Runs the npm command with the supplied args.
 
   @method npm
-  @param {String} command The npm command to run.
-  @param {Array} npmArgs The arguments passed to the npm command.
-  @param {Array} options The options passed when loading npm.
-  @param {Module} [npm] A reference to the npm module.
+  @param {Array} args The arguments passed to the npm command.
 */
-module.exports = function npm(command, npmArgs, options/*, npm*/) {
-  var lib;
-  if (arguments.length === 4) {
-    lib = arguments[3];
-  } else {
-    lib = require('npm');
-  }
-
-  var load = Promise.denodeify(lib.load);
-
-  return load(options)
-    .then(function() {
-      // if install is denodeified outside load.then(),
-      // it throws "Call npm.load(config, cb) before using this command."
-      var operation = Promise.denodeify(lib.commands[command]);
-
-      return operation(npmArgs || []);
+module.exports = function npm(args) {
+  return new Promise(function(resolve, reject) {
+    var npmPath = require.resolve('npm/bin/npm-cli.js');
+    debug.debug('npm binary resolved to: %s', npmPath);
+    if (!npmPath) {
+      return reject(new Error('NPM could not be found'));
+    }
+
+    var command = npmPath + ' ' + args.join(' ');
+    debug.info('%s', command);
+
+    exec(command, function(error) {
```

Comment:
since it's a child process, what happens when parent process dies (`ember install asdfasdfasdf` + `Ctrl + c`)? does it just linger there?